book/src/simple-payment-and-state-verification.md

```diff
@@ -0,0 +1,180 @@
+the block, and therefore, there is no way to infer from just the state whether a
+transaction that is encoded in the ledger actually succeeded in a state commit
+or failed.  It may not be necessary to encode the full code but a single
+status bit to indicate the transaction succeeded.
+
+### State Merkle Path
+
+A state receipt provides a receipt that a specific state is committed at 
+the end of the block.  Inter-block state transitions do not generate a receipt.
+
+For example:
+
+* A sends 5 lamports to B
+* B spends 5 lamports
+* C sends 5 lamports to A
+
+At the end of the block, A and B are in the exact same starting state, and any
+state receipt would point to the same value for A or B.
+
+The bank-merkle is computed from the merkle tree of the new state changes, along
+with the previous bank-merkle, and the block-merkle.
+ 
+```                
+                        bank-merkle 
+                   /                \
+                  /                  \
+          bank-diff-merkle        block-merkle
+                /       \
+               /         \
+            hash          previous bank-diff-merkle
+     /             \                   
+    /               \                   
+hash(Account1)  hash(Account2) 
+
+```
+
+A state receipt contains only the updated state changes for the block.  A direct
+Merkle path to the current bank-merkle guarantees the state value at that bank
+hash, but it cannot be used to generate a “current” receipt to the latest state
+if the state modification occurred in some previous block.  There is no
+guarantee that the path provided by the validator is the latest one available
+out of all the previous bank-merkles.
+
+Clients that want to query the chain for a receipt of the "latest" state would
+need to create a transaction that would update the Merkle path for that account,
+such as a credit of 0 lamports.
+
+###  Validator Votes
+
+Leaders should coalesce the validator votes by stake weight into a single entry.
+This will reduce the number of entries necessary to create a receipt.
+
+###  Chain of Entries
+
+A receipt is a PoH link from the payment or state Merkle path to a consecutive
+list of validation votes.
+
+
+It contains the following:
+* State -> Bank-Merkle
+or 
+* Transaction -> Entry-Merkle -> Block-Merkle -> Bank-Merkle
+
+And a vector of PoH entries:
+
+* Validator vote entries
+ 
+```
+* Ticks
+* Light entries 
+
+```
+/// This Entry definition skips over the transactions and only contains the
+/// hash of the transactions used to modify PoH.
+LightEntry { 
+    /// The number of hashes since the previous Entry ID.
+    pub num_hashes: u64,
+    /// The SHA-256 hash `num_hashes` after the previous Entry ID.
+    hash: Hash,
+    /// The Merkle root of the transactions encoded into the Entry.
+    entry_hash: Hash,
+}
+```
+
+The compressed entries are reconstructed from Entries and simply show the entry
```

Comment:
How frequently should a `LightEntry` be inserted into the ledger?

Reply:
It should replace the Entry which contain regular transactions. The hash allows the light client verify the PoH chain is still valid.

Reply:
Ok.  So somewhere in the code path including:
https://github.com/solana-labs/solana/blob/master/core/src/entry.rs#L179

And that `entry::hash_transactions()` above it is going to need to output a Merkle root instead of a hash over the concatenated tx sigs.